pkg/services/ngalert/api.go

```diff
@@ -148,36 +111,49 @@ func (ng *AlertNG) deleteAlertDefinitionEndpoint(c *models.ReqContext) api.Respo
 // updateAlertDefinitionEndpoint handles PUT /api/alert-definitions/:alertDefinitionId.
 func (ng *AlertNG) updateAlertDefinitionEndpoint(c *models.ReqContext, cmd updateAlertDefinitionCommand) api.Response {
 	cmd.ID = c.ParamsInt64(":alertDefinitionId")
-	cmd.SignedInUser = c.SignedInUser
-	cmd.SkipCache = c.SkipCache
+	cmd.OrgID = c.SignedInUser.OrgId
 
 	if err := ng.updateAlertDefinition(&cmd); err != nil {
 		return api.Error(500, "Failed to update alert definition", err)
 	}
 
-	return api.JSON(200, util.DynMap{"affectedRows": cmd.RowsAffected, "id": cmd.Result.Id})
+	return api.JSON(200, util.DynMap{"affectedRows": cmd.RowsAffected, "id": cmd.Result.ID})
```

Comment:
Nit nothing new. A bit weird to return `affectedRows`. Same applies for delete alert definition. At least if it's zero or greater than 1 something is wrong and should be treated as an error or something

Reply:
Zero is not exactly an error: it means that the alert definition does not exists, so in that case I could handled it and return 404 status. Do you think that this would be better?

Reply:
I realised that the zero case is already covered by the `validateOrgAlertDefinition` middleware [check](https://github.com/grafana/grafana/blob/a01ee33da70c71f7c32ea56da8a6526d9f6abf3c/pkg%2Fservices%2Fngalert%2Fmiddleware.go#L12) so it should be impossible the `RowsAffected ` to be something other than one. Therefore I have only added a [warning](https://github.com/grafana/grafana/blob/2c2a0479704fbc9ac43e6d4523690a262daef36c/pkg/services/ngalert/api.go#L137) for those cases.

Reply:
Definitely think 404 should be returned if it doesn't exist. In some places in the code base we let the sqlstore check rowsaffected and return `NotFoundError`, not sure if that's the proper way to go. With the changes returning the id of the entity affected is still a good thing to have I think instead of just a string. 

HTTP API design is tricky, but when implemented the folder API back in the days I was quite happy with returning the full DTO in the response ([code](https://github.com/grafana/grafana/blob/aa8fb1ae98bf25158d4d3431a8506d59718c35a5/pkg/api/folder.go#L83)), but highly personal.

These are nits and can be fixed later if you would like to get it merged.

Reply:
Just to clarify: `validateOrgAlertDefinition` returns 404 if the alert definition is not found.

Reply:
Thanks didn't notice that :+1: